emission/tests/modellingTests/modellingTestAssets.py

```diff
@@ -1,35 +1,145 @@
+    """
+    TODO: Write about each parameter to the function
+    pass in a test configuration to the binning algorithm.
+    
+    clustering_way : Part of the trip used for checking pairwise proximity.
+                        Can take one of the three values:
+                        
+                        1. 'origin' -> using origin of the trip to check if 2 points
+                                        lie within the mentioned similarity_threshold_meters
+                        2. 'destination' -> using destination of the trip to check if 2 points
+                                            lie within the mentioned similarity_threshold_meters
+                        3. 'origin-destination' -> both origin and destination of the trip to check 
+                                                if 2 points lie within the mentioned 
+                                                    similarity_threshold_meters
+    """        
+    model_config = {
+        "metric": metric,
+        "similarity_threshold_meters": threshold,  # meters,
+        "apply_cutoff": cutoff,
+        "clustering_way": clustering_way,  
+        "incremental_evaluation": incrementalevaluation
+    }
+
+    return eamtg.GreedySimilarityBinning(model_config)
+
+def generate_random_point():
+    """Generate a completetly random point valid WGS84 latitiude and longtidude"""
+    lat=random.uniform(-90,90)
+    lon=random.uniform(-180,180)
+    return [lat,lon]
+
+def generate_nearby_random_points(ref_coords,threshold):
+    """
+    Generate valid WGS84 latitiude and longtidude in threshold(m) proximity to
+    ref coordinates
+    """
+
+    thresholdInWGS84 = threshold* (0.000001/0.11)
+    dx=random.uniform(-thresholdInWGS84/2,thresholdInWGS84/2)
+    dy=random.uniform(-thresholdInWGS84/2,thresholdInWGS84/2)
+    return [ref_coords[0] +dx , ref_coords[1] +dy]
+
+def calDistanceTest(point1, point2, coordinates=False):
+    """haversine distance
+
+    :param point1: a coordinate in degrees WGS84
+    :param point2: another coordinate in degrees WGS84
+    :param coordinates: if false, expect a list of coordinates, defaults to False
+    :return: distance approximately in meters
+    """
+    earthRadius = 6371000  # meters
+    if coordinates:
+        dLat = math.radians(point1.lat-point2.lat)
+        dLon = math.radians(point1.lon-point2.lon)
+        lat1 = math.radians(point1.lat)
+        lat2 = math.radians(point2.lat)
+    else:
+        dLat = math.radians(point1[1]-point2[1])
+        dLon = math.radians(point1[0]-point2[0])
+        lat1 = math.radians(point1[1])
+        lat2 = math.radians(point2[1])
+
+
+    a = (math.sin(dLat/2) ** 2) + ((math.sin(dLon/2) ** 2) * math.cos(lat1) * math.cos(lat2))
+    c = 2 * math.atan2(math.sqrt(a), math.sqrt(1-a))
+    d = earthRadius * c
+
+    return d
+
+def setTripConfig(trips,trip_part,threshold,within_thr=None,label_data=None):
```

Comment:
this is super confusing because the name of the function is `setTripConfig`, but it has an argument called trips and it calls a function called `generate_mock_trip*s*` but then returns a variable called `trip`.
Also I am not sure why we need to wrap two lines of code in a separate function